src/main/java/tick/tasks/Task.java

```diff
@@ -0,0 +1,78 @@
+package tick.tasks;
+
+/**
+ * Task class represents a task with a description and a status that indicates whether the task is done.
+ */
+public abstract class Task {
+    private String description;
+    private boolean isDone;
+
+    /**
+     * Constructs a new Task with the specified description.
+     * The task is initially marked as not done.
+     *
+     * @param description Description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the status icon of the task.
+     * If the task is done, it returns "X"; otherwise, it returns a space.
+     *
+     * @return A string representing the status icon of the task.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Returns the description of the task.
+     *
+     * @return Description of the task.
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * Returns the status of the task.
+     *
+     * @return True if the task is done; false otherwise.
+     */
+    public boolean getStatus() {
+        return this.isDone;
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the task as not done.
+     */
+    public void markAsUndone() {
+        this.isDone = false;
+    }
+
+    /**
+     * Returns a string representation of the task that can be saved to a file.
+     *
+     * @return A string representation of the task that can be saved to a file.
+     */
+    public abstract String toFileString();

```

Comment:
the name `toFileString` might cause some confusion. Perhaps a more descriptive name would be `toStorageFormat`